xmake/rules/c++/modules/modules_support/gcc.lua

```diff
@@ -205,14 +200,14 @@ function generate_dependencies(target, sourcebatch, opt)
                 common.fallback_generate_dependencies(target, jsonfile, sourcefile, function(file)
                     local compinst = target:compiler("cxx")
                     local compflags = compinst:compflags({sourcefile = file, target = target})
-                    local flags = {}
-                    for _, flag in ipairs(compflags) do
-                        if flag:startswith("-std") or (flag:startswith("-f") and not flag:startswith("-fmodules")) or flag:startswith("-D") or flag:startswith("-U") or flag:startswith("-I") or flag:startswith("-isystem") then
-                            table.insert(flags, flag)
+                    for i, flag in ipairs(compflags) do
+                        -- exclude -fmodule* flags because, when they are set gcc try to find bmi of imported modules but they don't exists a this point of compilation
+                        if flag:startswith("-fmodule") then
+                            table.remove(compflags, i)
```

Comment:
it's non-safe. compflags is being traversed. you should use a new copy.

or use `table.remove_if`

```
function test_remove_if(t)
    t:are_equal(table.remove_if({1, 2, 3, 4, 5, 6}, function (i, v) return (v % 2) == 0 end), {1, 3, 5})
    t:are_equal(table.remove_if({a = 1, b = 2, c = 3}, function (i, v) return (v % 2) == 0 end), {a = 1, c = 3})
end
```